packages/block-editor/src/store/selectors.js

```diff
@@ -539,18 +539,10 @@ export const getBlockParents = createSelector(
 export const getBlockParentsByBlockName = createSelector(
 	( state, clientId, blockName, ascending = false ) => {
 		const parents = getBlockParents( state, clientId, ascending );
-		return parents
-			.map( ( id ) => ( {
-				id,
-				name: getBlockName( state, id ),
-			} ) )
-			.filter( ( { name } ) => {
-				if ( Array.isArray( blockName ) ) {
-					return blockName.includes( name );
-				}
-				return name === blockName;
-			} )
-			.map( ( { id } ) => id );
+		const hasName = Array.isArray( blockName )
+			? ( name ) => blockName.includes( name )
+			: ( name ) => blockName === name;
+		return parents.filter( ( id ) => hasName( getBlockName( state, id ) ) );
```

Comment:
Do we really need to redeclare the function on every run? Also, I find the polymorphic behavior of `hasName` a bit confusing - on one run it will accept an array only, and on another - a string only, while it has the same name. I think this gets behavior exposed better if we have a helper like this:

```
function hasBlockName( blockNames, name ) {
	if ( Array.isArray( blockNames ) ) {
		return blockNames.includes( name );
	}
	return blockNames === name;
}
```

and then use that inside `getBlockParentsByBlockName()`:

```
export const getBlockParentsByBlockName = createSelector(
	( state, clientId, blockName, ascending = false ) => {
		const parents = getBlockParents( state, clientId, ascending );
		return parents.filter( ( id ) =>
			hasBlockName( blockName, getBlockName( state, id ) )
		);
	},
	( state ) => [ state.blocks.parents ]
);
```

That way we also won't need to redeclare it on every run. WDYT?

Reply:
My intent was to avoid calling `Array.isArray` on every iteration of the `.filter` function. `blockName` is constant through that loop, so why not do the array check only once at the beginning?

Reply:
Yes, it's more an aesthetic concern rather than real perfomance, as the lists of `blockNames` and `parents` are always short. If you have a suggestion that makes the code more readable, and still keeps the single `isArray` check, I would love to see it!

Reply:
I can think of a few alternative ways, but they still sacrifice readability a bit. 

With that in mind, I think your approach is just fine 👍